babel.config.js

```diff
@@ -16,10 +16,12 @@ const webpack = {
         production: {
             presets: defaultPresets,
             plugins: [...defaultPlugins, 'transform-remove-console'],
+            sourceType: 'unambiguous',
         },
         development: {
             presets: defaultPresets,
             plugins: defaultPlugins,
+            sourceType: 'unambiguous',
```

Comment:
Why this change? Shouldn't this be `module` which is the default?

https://babeljs.io/docs/en/options#sourcetype

Reply:
Hi @marcaaron ,

I've got the solution from this link:

https://stackoverflow.com/a/56283408/1174657

In Win 10, this is what happened without unambiguous sourceType:

```
global.js:20 Uncaught TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
    at Module.<anonymous> (global.js:20)
    at Module../node_modules/webpack/buildin/global.js (app-3177bbb0b1e5efe731e1.bundle.js:159668)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
    at Module../node_modules/sockjs-client/dist/sockjs.js (app-3177bbb0b1e5efe731e1.bundle.js:152551)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
    at Object../node_modules/webpack-dev-server/client/clients/SockJSClient.js (SockJSClient.js:28)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
```

This link state that this error occurred when we are trying to use CommonJS while we need to use ES modules.

The alternative solution is by replacing 

```
module.exports = myfunction
```

with

```
export default myfunction
```

But I'm not sure which code to replace.

https://flaviocopes.com/cannot-assign-readonly-property-export/

Reply:
My guess is that it's the export at the bottom of this file and we can maybe change this to

```js
export default ({caller}) => {
    // For `react-native` (iOS/Android) caller will be "metro"
    // For `webpack` (Web) caller will be "@babel-loader"
    const runningIn = caller(({name}) => name);
    return ['metro', 'babel-jest'].includes(runningIn) ? metro : webpack;
}
```

Reply:
Ok I think in this case the best thing to do would be to leave a comment about why we need `sourceType: unambiguous` so that it is not altered in the future.

Many of our engineers and contributors use macOS so it is very likely that this will be accidentally removed without documentation. What do you say? Thanks.